common/server_protocolhandler.cpp

```diff
@@ -373,6 +390,22 @@ void Server_ProtocolHandler::pingClockTimeout()
 
     if (timeRunning - lastDataReceived > server->getMaxPlayerInactivityTime())
         prepareDestroy();
+ 
+    if (QString::fromStdString(userInfo->privlevel()).toLower() == "none")
+        if ((server->getIdleClientTimeout() > 0) && (idleClientWarningSent == true))
+            if (timeRunning - lastActionReceived > server->getIdleClientTimeout())
+                prepareDestroy();
+
+    if (QString::fromStdString(userInfo->privlevel()).toLower() == "none")
+        if (((timeRunning - lastActionReceived) >= ceil(server->getIdleClientTimeout() *.9)) && (idleClientWarningSent == false)) {
```

Comment:
`&& (!idleClientWarningSent)` would be a bit cleaner; comparing bool values to bool constants to get a resulting bool value is a bit excessive ;)